test/snac/data/ConstellationTest.php

```diff
@@ -330,6 +330,82 @@ public function testNotEqualsNonEmpty() {
 
     }
 
+    /**
+     * Test that equals() can ignore subcomponents
+     */
+     public function testEqualsCanIgnoreMetadata() {
+         $id1 = new \snac\data\Constellation();
+         $id2 = new \snac\data\Constellation();
+         $jsonIn = file_get_contents("test/snac/data/json/constellation_test.json");
+         $id1->fromJSON($jsonIn);
+         $id2->fromJSON($jsonIn);
+
+         $data = array("dataType"   => "SNACControlMetadata",
+                       "sourceData" => "University of Virginia",
+                       "language"   => "Spanish",
+                       "note"       => "This is a mocked metadata");
+
+         $metadata = new \snac\data\SNACControlMetadata($data);
+
+         $id1->addSNACControlMetadata($metadata);
+         $this->assertFalse($id1->equals($id2), "Equals failed to compare SNACControlMetadata by default");
+         $this->assertTrue($id1->equals($id2, false, false), "Equals failed to ignore SNACControlMetadata");
+     }
+
+     public function testEqualsCanIgnoreContributors() {
+         $id1 = new \snac\data\Constellation();
+         $id2 = new \snac\data\Constellation();
+         $jsonIn = file_get_contents("test/snac/data/json/constellation_test.json");
+         $id1->fromJSON($jsonIn);
+         $id2->fromJSON($jsonIn);
+         $name1 = new \snac\data\NameEntry();
+         $name2 = new \snac\data\NameEntry();
+
+         $nameData1 = array("dataType" => "NameEntry",
+                            "contributors" => [[ "dataType" => "Contributor", "name" => "Original"]]);
+
+         $nameData2 = array("dataType" => "NameEntry",
+                            "contributors" => [[ "dataType" => "Contributor", "name" => "Different"]]);
+
+        $name1->fromArray($nameData1);
+        $name2->fromArray($nameData2);
+        $id1->addNameEntry($name1);
+        $id2->addNameEntry($name2);
+
+        //should be inequal with $stict and $checkSubcomponents
```

Comment:
Looking too hard; I found a typo ;).  I think these tests are good.  Can you add an assertFalse on equals(.., false, true)?  So, not strict but do look at subcomponents?